### ISSUE TITLE
Sensor for itemspecific capacity

### ISSUE THREAD

**Issue by AmosNico** [Anuken/Mindustry-Suggestions#2971]
### Describe the content or mechanics you are proposing.

The opportunity in mlogic to get the capacity of a linked building for a specific item. For example:
"sensor leadcapacity factory1 @lead" should give 40 for a groundfactory with nova configuration and 0 for a crawler
"sensor leadcapacity crucible1 @lead" should give 0 for a silicon crucible

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

This should make it a lot easier to refill especially factories (but also factories) where before you had write lots of code to run through all unit-options and items that belong to them. In combination with the lookup funtion this would simplify this drastically.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by oinkymoo** [comment]
cool idea but it should definitely be called something else. `@lead` is already the thing you can sense for the amount currently inside

**Comment by AmosNico** [comment]
> cool idea but it should definitely be called something else. `@lead` is already the thing you can sense for the amount currently inside

I think of `@lead` as an object representing lead in logic that can be used for multiple purposes (sensor type config unloader1 for example can also return lead and the lookup function too). By by using this syntax (or something similar using `@lead`) it is compatible in combination with these other functions

**Comment by AmosNico** [comment]
> 
> 
> cool idea but it should definitely be called something else. `@lead` is already the thing you can sense for the amount currently inside

You're totally right, I meant to use 2 arguments to specify what should be sensed, but I forgot one, I fixed it now.

**Comment by AmosNico** [comment]
> sensor currently does not have a second argument, it would almost certainly have to be `@leadCapacity` or `@capacityLead`, and would most likely have a similar menu to the item amounts picker separate from the main sensable properties list

I'm not sure if this would add enough gameplay for the complexity it's adding. I don't really like the idea of adding an additional lookup. It adds unnecessary complexity to logic and doesn't make for example the result of a configuration sensor of an unloader compatible with the `@leadcapacity`. But I don't think that it would be that difficult to add a second argument to sensor, while keeping existing logic programs run. I noticed that for some of the statements it doesn't matter if you ommit ending arguments that are not used by that particular statement. For example `ucontrol flag 1` is accepted by mindustry, although ucontrol normally has many more arguments.
